File: LLM-Translations/GPT-4/codenet/C/Go/s445709355.go

package main

import (
	"fmt"
)

func in() int {
	var n int
	fmt.Scan(&n)
	return n
}

func calc(n int, f1, f2 []int) {
	k := n % 10
	for n >= 10 {
		n /= 10
	}
	f1[10*k+n]++
	f2[n*10+k]++
}

func main() {
	var N, ans int

	N = in()
	f1 := make([]int, 100)
	f2 := make([]int, 100)

	for N > 0 {
		calc(N, f1, f2)
		N--
	}

	ans = 0
	for i := 0; i < 100; i++ {
		ans += f1[i] * f2[i]
	}
	fmt.Printf("%d\n", ans)
}

File: LLM-Translations/GPT-4/codenet/C/Python/s445709355.py
def calc(n):
    k = n % 10
    while n >= 10:
        n //= 10
    f1[10*k + n] += 1
    f2[n*10 + k] += 1

def main():
    N = int(input())
    global f1, f2
    f1, f2 = [0] * 100, [0] * 100
    while N:
        calc(N)
        N -= 1

    ans = 0
    for i in range(100):
        ans += f1[i] * f2[i]
    print(ans)

if __name__ == "__main__":
    main()


